core/src/message/context.rs

```diff
@@ -291,3 +293,218 @@ impl MessageContext {
+        assert_eq!(
+            context.outcome.reply.as_ref().unwrap().payload().to_vec(),
+            vec![1, 2, 3, 0, 0],
+        );
+
+        // Checking that repeated call `reply_commit(...)` returns error and does not
+        assert!(context.reply_commit(reply_packet).is_err());
+
+        // Checking that at this point vector of outgoing messages is empty
+        assert!(context.outcome.handle.is_empty());
+
+        // Creating an expected handle for a future initialized message
+        let expected_handle = 0;
+
+        // Initializing message and compare its handle with expected one
+        assert_eq!(
+            context.send_init().expect("Error initializing new message"),
+            expected_handle
+        );
+
+        // And checking that it is not formed
+        assert!(context
+            .store
+            .outgoing
+            .get(&expected_handle)
+            .expect("This key should be")
+            .is_some());
+
+        // Checking that we are able to push payload for the
+        // message that we have not committed yet
+        assert!(context.send_push(expected_handle, &[5, 7]).is_ok());
+        assert!(context.send_push(expected_handle, &[9]).is_ok());
+
+        // Creating an outgoing packet to commit sending by parts
+        let commit_packet = HandlePacket::default();
+
+        // Checking if commit is successful
+        assert!(context.send_commit(expected_handle, commit_packet).is_ok());
+
+        // Checking that we are **NOT** able to push payload for the message or
+        // commit it if we already committed it or directly pushed before
+        assert!(context.send_push(expected_handle, &[5, 7]).is_err());
+        assert!(context
+            .send_commit(expected_handle, HandlePacket::default())
+            .is_err());
+
+        // Creating a handle to push and do commit non-existent message
+        let expected_handle = 15;
+
+        // Checking that we also get an error when trying
+        // to commit or send a non-existent message
+        assert!(context.send_push(expected_handle, &[0]).is_err());
+        assert!(context
+            .send_commit(expected_handle, HandlePacket::default())
+            .is_err());
+
+        // Creating a handle to init and do not commit later
+        // to show that the message will not be sent
+        let expected_handle = 1;
+
+        assert_eq!(
+            context.send_init().expect("Error initializing new message"),
+            expected_handle
+        );
+        assert!(context.send_push(expected_handle, &[2, 2]).is_ok());
+
+        // Checking that reply message not lost and matches our initial
+        assert!(context.outcome.reply.is_some());
+        assert_eq!(
+            context.outcome.reply.as_ref().unwrap().payload(),
+            vec![1, 2, 3, 0, 0]
+        );
+
+        // Checking that on drain we get only messages that were fully formed (directly sent or committed)
```

Comment:
place dot for the end of the sentence of each comment please

Reply:
Done.